src/layout/mod.rs

```diff
@@ -1431,6 +1569,29 @@ impl<W: LayoutElement> Layout<W> {
         }
     }
 
+    pub fn new_workspace_for_output(
+        &mut self,
+        workspace_config: &WorkspaceConfig,
+        output: Option<Output>,
+        config: &Config,
+    ) {
+        let options = Rc::new(Options::from_config(config));
```

Comment:
Shouldn't this use `self.options`?

Reply:
I don't think so, because this function is called during reload, where `self.options` is derived from the old config, while `options` is derived from the new one. At least if I understood things correctly. If I didn't, then yeah, this should use `self.options`, and the `config` argument becomes unnecessary.